### PR TITLE
MM-47282 : Migrate "/auth_sso/" e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by joremysh**
#### Summary

Converting following spec files under /integration/auth_sso to Typescript

1. authentication_1_spec.js
2. authentication_3_spec.js
3. authentication_not_cloud_spec.js
4. authentication_2_spec.js
5. authentication_4_spec.js
6. hide_create_account_spec.js

#### Ticket Link

Fixes mattermost/mattermost-server#21292

#### Related Pull Requests

No

#### Screenshots

No UI changes.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/package-lock.json
Comment (AshishDhama) [inline e2e/cypress/package-lock.json:48]: Could you please revert this change?
Was it creating any issues for you?

Comment (joremysh) [reply e2e/cypress/package-lock.json]: Done.
It was added automatically by `npm install`.

**Comment by joremysh** [comment]
> /e2e-test

I cannot check the e2e test result. Could you help check with it?

File: e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts
Comment (mvitale1989) [inline e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts:20]: This removal seems to cause the consistent failure of the MM-T1769 test case (which is pretty weird imo).
But i see that [the apiAdminLogin TS function spec](https://github.com/mattermost/mattermost-webapp/blob/a3afe9fb2abaf3b9ed49dd1403cac3adb280fe83/e2e/cypress/tests/support/api/user.d.ts#L54) doesn't accept any parameter either, which i suppose is why this was removed in this PR.

@saturninoabril would you perhaps know why the adminLogin may fail here, making `failOnStatusCode=false` required here? Should the function signature be changed?

Comment (furqanmlk) [reply e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts:20]: @mvitale1989 
remove both `cy.apiAdminLogin()` and `cy.apiUpdateConfig({})` from `after` block.
Passing all three for me. 

However, double check `cy.focused().tab();` line, it sometimes does not perform `tab` properly
 
image

File: e2e/cypress/tests/support/ui/account_settings_modal.d.ts
Comment (furqanmlk) [inline e2e/cypress/tests/support/ui/account_settings_modal.d.ts:30]:
```suggestion
        uiOpenProfileModal(section?: string): Chainable;
```
Can you also update return type instead of `any`
maybe `uiOpenProfileModal(section: string = ''): Chainable<Jquery>;`
or 
`uiOpenProfileModal(section: string = ''): Chainable<Element>;`
depends on definition

**Comment by furqanmlk** [comment]
@joremysh 
Please re-request reviewers when you are done with fixes :)

image

File: e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts
Comment (joremysh) [reply e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts:20]: should I remove `cy.apiAdminLogin();` and `cy.apiUpdateConfig({});` in `after` block?
It seems the e2e tests for this part requires an enterprise license so I could not run them on my machine.

File: e2e/cypress/tests/support/ui/account_settings_modal.d.ts
Comment (joremysh) [reply e2e/cypress/tests/support/ui/account_settings_modal.d.ts]: is the return type `Chainable<void>`?
image

File: e2e/cypress/tests/support/ui/account_settings_modal.d.ts
Comment (furqanmlk) [inline e2e/cypress/tests/support/ui/account_settings_modal.d.ts:30]:
```suggestion
        uiOpenProfileModal(section?: string): Chainable<JQuery<HTMLElement>>;
```

uiOpenProfileModal is returning HTML Element.

**Review by furqanmlk — COMMENTED** [review]
Thanks @joremysh 
All tests passed 
Just one small change.

File: e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts
Comment (mvitale1989) [reply e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts:20]: Thanks for the recommendation @furqanmlk!
Regarding the `cy.focused().tab();` line, I also see it's somewhat unreliable. But I could not reproduce the issue manually so I suppose it only happens because of a race condition in the UI automation.

@joremysh could you please change that line to `cy.focused().blur();` instead? It seems to be more reliable than emulating the tab keypress. Thank you!

**Review by furqanmlk — APPROVED** [review]
Thanks @joremysh 
LGTM !

**Comment by mvitale1989** [comment]
Thanks @joremysh, I have only one [last small suggestion](https://github.com/mattermost/mattermost-webapp/pull/11365/files#r1006104755) to improve the test reliability.
But other than that this looks good now, and I'd be ready to merge after that comment is addressed!

File: e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts
Comment (joremysh) [reply e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts:20]: updated

Comment (mvitale1989) [reply e2e/cypress/tests/integration/auth_sso/authentication_3_spec.ts:20]: Thanks!

**Review by mvitale1989 — APPROVED** [review]

**Comment by joremysh** [comment]
@mvitale1989 is there any action needed from me before merging the PR?

**Comment by mvitale1989** [comment]
No, all looks good now. Thank you for the contribution! :beers:

**Pull request merged**
